src/components/react-json-schemaForm/GeometryField/MapInteraction.js

```diff
@@ -0,0 +1,269 @@
+import { memo, useCallback, useEffect, useMemo, useContext } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import {
+  CONTROL_DRAW,
+  CONTROL_PATH,
+  CONTROLS_TOP_LEFT,
+} from '@terralego/core/modules/Map';
+import PropTypes from 'prop-types';
+import { getCoordinatesFromGeometries, getDirectionsThemes, getLayerId } from './utils';
+
+import { getLayers } from '../../../modules/CRUD/services/CRUD';
+
+import { CRUDContext } from '../../../modules/CRUD/services/CRUDProvider';
+import { MapContext } from '../../../modules/CRUD/services/MapProvider';
+
+import {
+  ALL,
+  POINT,
+  LINESTRING,
+  POLYGON,
+  MULTI_POINT,
+  MULTI_LINESTRING,
+  MULTI_POLYGON,
+} from '../../../utils/geom';
+
+
+import './styles.scss';
+
+const MapInteraction = ({
+  geomValues,
+  featuresToFitBounds,
+  setFeaturesToFitBounds,
+  setFormData,
+}) => {
+  const {
+    i18n: {
+      language,
+    },
+  } = useTranslation();
+
+  const { settings } = useContext(CRUDContext);
+
+  const {
+    addControl,
+    map,
+    removeControl,
+    setFitBounds,
+  } = useContext(MapContext);
+
+  const layers = useMemo(() => getLayers(settings), [settings]);
+
+  const resetStyle = useCallback(() => {
+    map && map.setFilter(getLayerId(layers, geomValues.layerName), null);
+  }, [geomValues.layerName, layers, map]);
+
+  const updateGeometryFromMap = useCallback(
+    ({ target, ...props }) => {
+      if (target.pathControl) {
+        const { geometry } = target.pathControl.getLineString();
+        setFormData(geometry);
+        return;
+      }
+
+      let { features } = target.draw.getAll();
+
+      const isMultiGeometry = ![POINT, LINESTRING, POLYGON].includes(geomValues.geomType);
+      if (features.length > 1 && !isMultiGeometry) {
+        const { features: [{ id }] } = props;
+        const { currentFeature, otherFeaturesIds } = features.reduce(
+          (list, item) => {
+            if (item.id === id) {
+              list.currentFeature.push(item);
+            } else {
+              list.otherFeaturesIds.push(item.id);
+            }
+            return list;
+          },
+          { currentFeature: [], otherFeaturesIds: [] },
+        );
+
+        target.draw.delete(otherFeaturesIds);
+        features = currentFeature;
+      }
+      setFormData({
+        type: features[0].geometry.type,
+        coordinates: getCoordinatesFromGeometries(features, isMultiGeometry),
+      });
+    },
+    [geomValues.geomType, setFormData],
+  );
+
+  useEffect(() => {
+    if (!map) {
+      return () => null;
+    }
+    const {
+      geom: geometry,
+      identifier,
+      layerName,
+      geomType,
+      properties,
+      routingSettings,
+    } = geomValues;
+
+    const layerId = getLayerId(layers, layerName);
+
+    const {
+      config: {
+        default: {
+          map: { accessToken },
+        },
+      },
+    } = settings;
+
+    const routingControl = {
```

Comment:
Can you create functions to initialize this two values (othersControl) outside of the component ?

Reply:
Done in the component (there are to much props dependence to be relevant outside)